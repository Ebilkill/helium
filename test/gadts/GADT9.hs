data Expr a where
    I :: Int -> Expr Int
    B :: Bool -> Expr Bool
    Plus :: Num a => Expr a -> Expr a -> Expr a

eval :: Expr a -> a
eval (I i) = i
eval (B b) = b
eval (Plus x y) = eval x + eval y