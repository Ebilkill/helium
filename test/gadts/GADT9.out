Prelude is up to date
Compiling gadts/GADT9.hs
fromList [(eval,Expr a -> a)]

Length of errors doesn't match

fromList [(eval,Expr a -> a)]
"fromList [(eval,Expr a -> a)]"
(8,7): Type error in pattern application
 pattern          : B b
 constructor      : B
   type           : Bool -> Expr Bool
   does not match : Int  -> Expr Int 

Compilation failed with 1 error
